### ISSUE TITLE
TF to TFLite conversion clarification

### ISSUE THREAD

**Issue by sandeep22-v** [tensorflow/tensorflow#48632]
**What is  the difference between 

converter.target_spec.supported_ops =[tf.lite.OpsSet.TFLITE_BUILTINS_INT8]

and 

converter.target_spec.supported_ops =[tf.lite.OpsSet.TFLITE_BUILTINS]

performed during conversion of tf model to tflite model??**

I tried using converter.target_spec.supported_ops in 3 ways : 
1.  using 
converter.target_spec.supported_ops =[tf.lite.OpsSet.TFLITE_BUILTINS]

2. using 
converter.target_spec.supported_ops =[tf.lite.OpsSet.TFLITE_BUILTINS_INT8]

3. execeuting code without converter.target_spec.supported_ops

when I was converting pb model to 8 bit quantized tflite model.

All three of them  yielded me the same results in the output tflite model,which were identical in all respects.
**Why is this so?**

**Comment by abattery** [comment]
@liufengdb could you triage this issue?

**Comment by sandeep22-v** [comment]
> @liufengdb could you triage this issue?

Sure sir...
I was converting a custom trained pb model to an 8 bit integer quantized tflite model.
That's where the question arose in my mind about this line : 
converter.target_spec.supported_ops 

Although I experimented with it and generated multiple tflite models,all of them were identical .
Also I cam across an article which stated that : 


**type_choice['none'] = None**   
**# for converter.target_spec.supported_ops**
**ops_choice = {**
    **"int8": [lite.OpsSet.TFLITE_BUILTINS_INT8],**
    **"tflite": [lite.OpsSet.TFLITE_BUILTINS],  # default**
    **"tf": [lite.OpsSet.SELECT_TF_OPS, lite.OpsSet.TFLITE_BUILTINS]**  
**}**

So basically,I got 3 identical tflite models with 3 configurations,whereas this article states otherwise

So sir can you please tell me the significance of 
**converter.target_spec.supported_ops** and why the 3 configurations I mentioned in the first post had yielded me identical tflite models??

**Comment by liufengdb** [comment]
The 'converter.target_spec.supported_ops' is just the inference the tflite converter exposed to the end users, and internally this flag will be used to specify some parameters to the backend for whether "allowing" floating-point ops in a quantized model and whether "allowing" flex ops, etc. "Allowing" certain ops doesn't necessarily mean the ops will be used. The converter will still try its best to use the most efficient ops.
Specifically, "lite.OpsSet.TFLITE_BUILTINS" is a super-set of "tf.lite.OpsSet.TFLITE_BUILTINS_INT8". If your model can be fully quantized, the converter will use the int8 ops, even though floating-point ops is allowed. So it is not surprised to see the results are the same for the three flag settings.
